BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c

```diff
@@ -208,6 +217,75 @@ AcpiTableUpdate (
   while (Size < Length) {
     AcpiHdr = (EFI_ACPI_DESCRIPTION_HEADER *)(AcpiTable + Size);
 
+    // Update SBL Performance Table if signature matches
+    if (AcpiHdr->Signature == EFI_ACPI_5_0_FIRMWARE_PERFORMANCE_DATA_TABLE_SIGNATURE) {
+      // FPDT points to SBL Performance Table
+      Fpdt = (FIRMWARE_PERFORMANCE_TABLE *)AcpiHdr;
+      SblPerfTable = (SBL_PERFORMANCE_TABLE *)(UINTN)Fpdt->SblPerfPointerRecord.SblPerfTablePointer;
+      if (SblPerfTable == NULL) {
+        DEBUG((DEBUG_INFO, "SBL Performance table not found!\n"));
+        Status = EFI_NOT_FOUND;
+        break;
+      } else {
+        DEBUG ((DEBUG_INFO, "SBLT found @ 0x%X, updating..\n", SblPerfTable));
+      }
+
+      // Get performance data from HOB
+      GuidHob = GetNextGuidHob (&gLoaderPerformanceInfoGuid, GetHobList());
+      if (GuidHob == NULL) {
+        ASSERT (FALSE);
+        Status = EFI_NOT_FOUND;
+        break;
+      }
+      PerfData = (PERFORMANCE_INFO *)GET_GUID_HOB_DATA (GuidHob);
+
+      // Grab relevant performance metrics
+      for (PerfIdx = 0; PerfIdx < MAX_TS_NUM; PerfIdx++) {
+        Id   = (RShiftU64 (PerfData->TimeStamp[PerfIdx], 48)) & 0xFFFF;
+        switch (Id) {
+        case 0x1000:  // Reset vector time
+        PerfTsc = PerfData->TimeStamp[PerfIdx] & 0x0000FFFFFFFFFFFFULL;
+        Time = (UINT32)DivU64x32 (PerfTsc, PerfData->Frequency);
+        ResetVectorTime = Time;
+        break;
+        case 0x3000:  // Stage 1 done (Stage 2 entry)
+          PerfTsc = PerfData->TimeStamp[PerfIdx] & 0x0000FFFFFFFFFFFFULL;
```

Comment:
Change this to a routine instead

Reply:
Thanks, will change to a routine and push